actions/login-to-gcs/action.yaml

```
@@ -0,0 +1,62 @@
+name: Login to GCS
+description: Composite action to login to Google Cloud Storage
+inputs:
+  bucket:
+    description: |
+      Name of bucket to upload to. Will default to grafanalabs-${repository.name}-${environment}
+    default: ""
+  environment:
+    description: |
+      Environment for uploading objects (can be either dev or prod).
+    default: dev
+
+outputs:
+  bucket:
+    description: "The name of the bucket that we have authenticated to."
+    value: ${{ steps.get-bucket-name.outputs.bucket }}
+
+runs:
+  using: composite
+  steps:
+    - name: Resolve GCP project
+      id: resolve-project
+      shell: bash
+      run: |
+        if [ "${{ inputs.environment }}" = "dev" ]; then
+          PROJECT="grafanalabs-dev"
+        elif [ "${{ inputs.environment }}" == "prod" ]; then
+          PROJECT="grafanalabs-global"
+        else
+          echo "Invalid environment. Valid environment variable inputs: dev, prod"
+          exit 1
+        fi
+        echo "project=${PROJECT}" >> ${GITHUB_OUTPUT}
```

also a nit, something that I've found useful:

```suggestion
        echo "project=${PROJECT}" | tee -a ${GITHUB_OUTPUT}
```

(also for the other outputs). It means the variables get logged which can be helpful sometimes.